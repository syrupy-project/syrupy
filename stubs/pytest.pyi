from typing import (
    Any,
    Callable,
    TypeVar,
)

ReturnType = TypeVar("ReturnType")

def fixture(func: Callable[..., ReturnType]) -> Callable[..., ReturnType]: ...

class Function: ...
class Item: ...
