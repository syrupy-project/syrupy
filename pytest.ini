[pytest]
addopts = -p pytester -p syrupy
