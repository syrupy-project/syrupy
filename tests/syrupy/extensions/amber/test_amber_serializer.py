from collections import namedtuple

import pytest


def test_non_snapshots(snapshot):
    with pytest.raises(AssertionError):
        assert "Lorem ipsum." == "Muspi merol."


def test_reflection(snapshot):
    assert snapshot == snapshot


def test_empty_snapshot(snapshot):
    assert snapshot == None  # noqa: E711
    assert snapshot == ""


def test_newline_control_characters(snapshot):
    assert snapshot == "line 1\nline 2"
    assert snapshot == "line 1\r\nline 2"
    assert snapshot == "line 1\r\nline 2\r\n"
    assert snapshot == "line 1\rline 2\r"
    assert snapshot == "line 1\rline 2\n"
    assert snapshot == "line 1\rline 2"


def test_multiline_string_in_dict(snapshot):
    lines = "\n".join(["line 1", "line 2"])
    assert {"value": lines} == snapshot


def test_deeply_nested_multiline_string_in_dict(snapshot):
    lines = "\n".join(["line 1", "line 2", "line 3"])
    d = {"value_a": {"value_b": lines}}
    assert d == snapshot


@pytest.mark.parametrize("actual", [False, True])
def test_bool(actual, snapshot):
    assert actual == snapshot


@pytest.mark.parametrize(
    "actual",
    [
        "",
        r"Raw string",
        r"Escaped \n",
        r"Backslash \u U",
        "ü•ûüêçüçØ",
        "singleline:",
        "- singleline",
        "multi-line\nline 2\nline 3",
        "multi-line\nline 2\n  line 3",
        "string with 'quotes'",
        b"Byte string",
    ],
    ids=lambda x: "",
)
def test_string(snapshot, actual):
    assert snapshot == actual


def test_multiple_snapshots(snapshot):
    assert "First." == snapshot
    snapshot.assert_match("Second.")
    assert snapshot == "Third."


ExampleTuple = namedtuple("ExampleTuple", ["a", "b", "c", "d"])


def test_tuple(snapshot):
    assert snapshot == ("this", "is", ("a", "tuple"))
    assert snapshot == ExampleTuple(a="this", b="is", c="a", d={"named", "tuple"})
    assert snapshot == ()


@pytest.mark.parametrize(
    "actual",
    [
        {"this", "is", "a", "set"},
        {"contains", "frozen", frozenset({"1", "2"})},
        {"contains", "tuple", (1, 2)},
        {"contains", "namedtuple", ExampleTuple(a=1, b=2, c=3, d=4)},
        set(),
    ],
)
def test_set(snapshot, actual):
    assert snapshot == actual


@pytest.mark.parametrize(
    "actual",
    [
        {"b": True, "c": "Some text.", "d": ["1", 2], "a": {"e": False}},
        {"b": True, "c": "Some ttext.", "d": ["1", 2], "a": {"e": False}},
        {
            1: True,
            "a": "Some ttext.",
            frozenset({"1", "2"}): ["1", 2],
            ExampleTuple(a=1, b=2, c=3, d=4): {"e": False},
        },
        {},
        {"key": ["line1\nline2"]},
        {"key": [1, "line1\nline2", 2, "line3\nline4"]},
        {"key": [1, ["line1\nline2"], 2]},
    ],
)
def test_dict(snapshot, actual):
    assert actual == snapshot


def test_numbers(snapshot):
    assert snapshot == 3.5
    assert snapshot == 7
    assert snapshot == 2 / 6


@pytest.mark.parametrize(
    "actual",
    [
        [],
        ["this", "is", "a", "list"],
        ["contains", "empty", []],
        [1, 2, "string", {"key": "value"}],
    ],
)
def test_list(snapshot, actual):
    assert actual == snapshot


list_cycle = [1, 2, 3]
list_cycle.append(list_cycle)

dict_cycle = {"a": 1, "b": 2, "c": 3}
dict_cycle.update(d=dict_cycle)


@pytest.mark.parametrize("cyclic", [list_cycle, dict_cycle])
def test_cycle(cyclic, snapshot):
    assert cyclic == snapshot


class CustomClass:
    a = 1
    b = "2"
    c = list_cycle
    d = dict_cycle
    _protected_variable = None
    __private_variable = None

    def __init__(self, x=None):
        self.x = x
        self._y = 1
        self.__z = 2

    def public_method(self, a, b=1, *, c, d=None):
        pass

    def _protected_method(self):
        pass

    def __private_method(self):
        pass


def test_custom_object_repr(snapshot):
    assert CustomClass(CustomClass()) == snapshot


class TestClass:
    def test_class_method_name(self, snapshot):
        assert snapshot == "this is in a test class"

    @pytest.mark.parametrize("actual", ["a", "b", "c"])
    def test_class_method_parametrized(self, snapshot, actual):
        assert snapshot == actual

    @pytest.mark.parametrize("actual", ["x", "y", "z"])
    class TestNestedClass:
        def test_nested_class_method(self, snapshot, actual):
            assert snapshot == f"parameterized nested class method {actual}"


class TestSubClass(TestClass):
    pass


@pytest.mark.parametrize("parameter_with_dot", ("value.with.dot",))
def test_parameter_with_dot(parameter_with_dot, snapshot):
    assert parameter_with_dot == snapshot


@pytest.mark.parametrize("parameter_1", ("foo",))
@pytest.mark.parametrize("parameter_2", ("bar",))
def test_doubly_parametrized(parameter_1, parameter_2, snapshot):
    assert parameter_1 == snapshot
    assert parameter_2 == snapshot
