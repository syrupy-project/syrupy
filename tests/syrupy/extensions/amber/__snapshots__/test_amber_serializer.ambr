# name: TestClass.TestNestedClass.test_nested_class_method[x]
  'parameterized nested class method x'
---
# name: TestClass.TestNestedClass.test_nested_class_method[y]
  'parameterized nested class method y'
---
# name: TestClass.TestNestedClass.test_nested_class_method[z]
  'parameterized nested class method z'
---
# name: TestClass.test_class_method_name
  'this is in a test class'
---
# name: TestClass.test_class_method_parametrized[a]
  'a'
---
# name: TestClass.test_class_method_parametrized[b]
  'b'
---
# name: TestClass.test_class_method_parametrized[c]
  'c'
---
# name: TestSubClass.TestNestedClass.test_nested_class_method[x]
  'parameterized nested class method x'
---
# name: TestSubClass.TestNestedClass.test_nested_class_method[y]
  'parameterized nested class method y'
---
# name: TestSubClass.TestNestedClass.test_nested_class_method[z]
  'parameterized nested class method z'
---
# name: TestSubClass.test_class_method_name
  'this is in a test class'
---
# name: TestSubClass.test_class_method_parametrized[a]
  'a'
---
# name: TestSubClass.test_class_method_parametrized[b]
  'b'
---
# name: TestSubClass.test_class_method_parametrized[c]
  'c'
---
# name: test_bool[False]
  False
---
# name: test_bool[True]
  True
---
# name: test_custom_object_repr
  <class 'CustomClass'> {
    a=1,
    b='2',
    c=<class 'list'> [
      1,
      2,
      3,
      ...,
    ],
    d=<class 'dict'> {
      'a': 1,
      'b': 2,
      'c': 3,
      'd': ...,
    },
    x=<class 'CustomClass'> {
      a=1,
      b='2',
      c=<class 'list'> [
        1,
        2,
        3,
        ...,
      ],
      d=<class 'dict'> {
        'a': 1,
        'b': 2,
        'c': 3,
        'd': ...,
      },
      x=None,
    },
  }
---
# name: test_cycle[cyclic0]
  <class 'list'> [
    1,
    2,
    3,
    ...,
  ]
---
# name: test_cycle[cyclic1]
  <class 'dict'> {
    'a': 1,
    'b': 2,
    'c': 3,
    'd': ...,
  }
---
# name: test_deeply_nested_multiline_string_in_dict
  <class 'dict'> {
    'value_a': <class 'dict'> {
      'value_b': '
        line 1
        line 2
        line 3
      ',
    },
  }
---
# name: test_dict[actual0]
  <class 'dict'> {
    'a': <class 'dict'> {
      'e': False,
    },
    'b': True,
    'c': 'Some text.',
    'd': <class 'list'> [
      '1',
      2,
    ],
  }
---
# name: test_dict[actual1]
  <class 'dict'> {
    'a': <class 'dict'> {
      'e': False,
    },
    'b': True,
    'c': 'Some ttext.',
    'd': <class 'list'> [
      '1',
      2,
    ],
  }
---
# name: test_dict[actual2]
  <class 'dict'> {
    '
      multi
      line
      key
    ': 'Some morre text.',
    'a': 'Some ttext.',
    1: True,
    <class 'ExampleTuple'> (
      a=1,
      b=2,
      c=3,
      d=4,
    ): <class 'dict'> {
      'e': False,
    },
    <class 'frozenset'> {
      '1',
      '2',
    }: <class 'list'> [
      '1',
      2,
    ],
  }
---
# name: test_dict[actual3]
  <class 'dict'> {
  }
---
# name: test_dict[actual4]
  <class 'dict'> {
    'key': <class 'list'> [
      '
        line1
        line2
      ',
    ],
  }
---
# name: test_dict[actual5]
  <class 'dict'> {
    'key': <class 'list'> [
      1,
      '
        line1
        line2
      ',
      2,
      '
        line3
        line4
      ',
    ],
  }
---
# name: test_dict[actual6]
  <class 'dict'> {
    'key': <class 'list'> [
      1,
      <class 'list'> [
        '
          line1
          line2
        ',
      ],
      2,
    ],
  }
---
# name: test_doubly_parametrized[bar-foo]
  'foo'
---
# name: test_doubly_parametrized[bar-foo].1
  'bar'
---
# name: test_empty_snapshot
  None
---
# name: test_empty_snapshot.1
  ''
---
# name: test_list[actual0]
  <class 'list'> [
  ]
---
# name: test_list[actual1]
  <class 'list'> [
    'this',
    'is',
    'a',
    'list',
  ]
---
# name: test_list[actual2]
  <class 'list'> [
    'contains',
    'empty',
    <class 'list'> [
    ],
  ]
---
# name: test_list[actual3]
  <class 'list'> [
    1,
    2,
    'string',
    <class 'dict'> {
      'key': 'value',
    },
  ]
---
# name: test_multiline_string_in_dict
  <class 'dict'> {
    'value': '
      line 1
      line 2
    ',
  }
---
# name: test_multiple_snapshots
  'First.'
---
# name: test_multiple_snapshots.1
  'Second.'
---
# name: test_multiple_snapshots.2
  'Third.'
---
# name: test_newline_control_characters
  '
  line 1
  line 2
  '
---
# name: test_newline_control_characters.1
  '
  line 1
  line 2
  '
---
# name: test_newline_control_characters.2
  '
  line 1
  line 2
  
  '
---
# name: test_newline_control_characters.3
  '
  line 1
  line 2
  '
---
# name: test_newline_control_characters.4
  '
  line 1
  line 2
  
  '
---
# name: test_newline_control_characters.5
  '
  line 1
  line 2
  '
---
# name: test_numbers
  3.5
---
# name: test_numbers.1
  7
---
# name: test_numbers.2
  0.3333333333333333
---
# name: test_parameter_with_dot[value.with.dot]
  'value.with.dot'
---
# name: test_reflection
  <class 'SnapshotAssertion'> {
    name='snapshot',
    num_executions=0,
  }
---
# name: test_set[actual0]
  <class 'set'> {
    'a',
    'is',
    'set',
    'this',
  }
---
# name: test_set[actual1]
  <class 'set'> {
    'contains',
    'frozen',
    <class 'frozenset'> {
      '1',
      '2',
    },
  }
---
# name: test_set[actual2]
  <class 'set'> {
    'contains',
    'tuple',
    <class 'tuple'> (
      1,
      2,
    ),
  }
---
# name: test_set[actual3]
  <class 'set'> {
    'contains',
    'namedtuple',
    <class 'ExampleTuple'> (
      a=1,
      b=2,
      c=3,
      d=4,
    ),
  }
---
# name: test_set[actual4]
  <class 'set'> {
  }
---
# name: test_snapshot_markers
  '
  # 
    # 
  ---
  # name:
  '
---
# name: test_string[0]
  ''
---
# name: test_string[10]
  b'Byte string'
---
# name: test_string[1]
  'Raw string'
---
# name: test_string[2]
  'Escaped \\n'
---
# name: test_string[3]
  'Backslash \\u U'
---
# name: test_string[4]
  'ü•ûüêçüçØ'
---
# name: test_string[5]
  'singleline:'
---
# name: test_string[6]
  '- singleline'
---
# name: test_string[7]
  '
  multi-line
  line 2
  line 3
  '
---
# name: test_string[8]
  '
  multi-line
  line 2
    line 3
  '
---
# name: test_string[9]
  "string with 'quotes'"
---
# name: test_tuple
  <class 'tuple'> (
    'this',
    'is',
    <class 'tuple'> (
      'a',
      'tuple',
    ),
  )
---
# name: test_tuple.1
  <class 'ExampleTuple'> (
    a='this',
    b='is',
    c='a',
    d=<class 'set'> {
      'named',
      'tuple',
    },
  )
---
# name: test_tuple.2
  <class 'tuple'> (
  )
---
